modules/extensions-xml-support/src/main/java/org/mule/runtime/extension/internal/ast/MacroExpandedComponentAst.java

```diff
@@ -107,14 +111,39 @@ public <T> Either<String, T> getValue() {
         }
 
         return (Either<String, T>) originalValue
-            .mapRight(v -> {
-              if (v instanceof String) {
-                return macroExpandedRawValue((String) v);
-              } else if (getDecorated().getModel().getName().equals(ERROR_MAPPINGS_PARAMETER_NAME)) {
-                return mapErrorMappings(getDecorated());
-              }
-              return v;
-            });
+            .mapRight(this::mapComponent);
+      }
+
+      private <T> Object mapComponent(T rawValue) {
+        if (isAnExpression(rawValue)) {
+          return macroExpandedRawValue((String) rawValue);
+        } else if (isAnErrorMappings()) {
+          return mapErrorMappings(getDecorated());
+        } else if (supportsChildDeclaration() && !isAKnownType(rawValue)) {
+          return copyMacroExpandedComponentTreeRecursively();
+        }
+        return rawValue;
+      }
+
+      private <T> boolean isAnExpression(T rawValue) {
```

Comment:
I would change isAnExpression -> mayBeAnExpression.
I would add the comments to the invocation of the methods which is where the reader of the code benefits from the comment.

Reply:
mmm, but from how thi is used, it looks that the value MUST be an expression, so the check for the markers would be missing here

Reply:
Rereading the code (Aye only added a method, didn't change what was originally there) I would name it must mustMacroExpandRawValue and not perform that verification. There is also some logic regarding adding a suffix if it is global which may not applie to the #[ ].